src/JACC.jl

```diff
@@ -33,17 +33,38 @@ export default_float
 export @atomic
 export parallel_for, parallel_reduce
 export shared
+export LaunchSpec
+export synchronize
 
-function parallel_for(
-        ::ThreadsBackend, N::I, f::F, x...) where {I <: Integer, F <: Function}
+synchronize(::ThreadsBackend) = nothing
+
+function default_stream end
+
+@kwdef mutable struct LaunchSpec{Backend}
+    stream = default_stream(Backend)
+    threads = 0
```

Comment:
Just curious why some are typed explicitly and some are not. FYI, this will default to Int64.

Reply:
the type of the stream is determined by the backend. Threads can be either an Integer or a tuple of 2 or 3 integers. same with blocks